src/Mutation/FileMutationGenerator.php

```diff
@@ -73,15 +73,15 @@ public function __construct(
      *
      * @throws UnparsableFile
      *
-     * @return Mutation[]
+     * @return iterable|Mutation[]
      */
     public function generate(
         SplFileInfo $fileInfo,
         bool $onlyCovered,
         LineCodeCoverage $codeCoverage,
         array $mutators,
         array $nodeIgnorers
-    ): array {
+    ): iterable {
```

Comment:
Here and below we use `iterable` typehint for generators, for one reason because it hides implementation details, and for another because it makes a lot easier to mock these methods: any array is an iterable, so one doesn't need to go through the hoops faking up generators.